components/calendar/src/hebrew.rs

```diff
@@ -0,0 +1,848 @@
+    }
+
+    fn months_in_year(&self, date: &Self::DateInner) -> u8 {
+        date.0.months_in_year()
+    }
+
+    fn days_in_year(&self, date: &Self::DateInner) -> u16 {
+        date.0.days_in_year()
+    }
+
+    fn days_in_month(&self, date: &Self::DateInner) -> u8 {
+        date.0.days_in_month()
+    }
+
+    fn offset_date(&self, date: &mut Self::DateInner, offset: DateDuration<Self>) {
+        date.0.offset_date(offset)
+    }
+
+    fn until(
+        &self,
+        date1: &Self::DateInner,
+        date2: &Self::DateInner,
+        _calendar2: &Self,
+        _largest_unit: DateDurationUnit,
+        _smallest_unit: DateDurationUnit,
+    ) -> DateDuration<Self> {
+        date1.0.until(date2.0, _largest_unit, _smallest_unit)
+    }
+
+    fn debug_name(&self) -> &'static str {
+        "Hebrew"
+    }
+
+    fn year(&self, date: &Self::DateInner) -> types::FormattableYear {
+        Self::year_as_hebrew(date.0.year)
+    }
+
+    fn month(&self, date: &Self::DateInner) -> types::FormattableMonth {
+        date.0.month()
+    }
+
+    fn day_of_month(&self, date: &Self::DateInner) -> types::DayOfMonth {
+        date.0.day_of_month()
+    }
+
+    fn day_of_year_info(&self, date: &Self::DateInner) -> types::DayOfYearInfo {
+        let prev_year = date.0.year.saturating_sub(1);
+        let next_year = date.0.year.saturating_add(1);
+        types::DayOfYearInfo {
+            day_of_year: date.0.day_of_year(),
+            days_in_year: date.0.days_in_year(),
+            prev_year: Self::year_as_hebrew(prev_year),
+            days_in_prev_year: Self::days_in_provided_year(prev_year),
+            next_year: Self::year_as_hebrew(next_year),
+        }
+    }
+}
+
+impl Hebrew {
+    /// Constructs a new Hebrew Calendar
+    pub fn new() -> Self {
+        Self
+    }
+
+    // Hebrew New Moon
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2244
+    #[allow(dead_code)]
+    pub(crate) fn molad(h_year: i32, h_month: u8) -> Moment {
+        let y = if h_month < TISHRI { h_year + 1 } else { h_year };
+
+        let months_elapsed = (h_month as f64 - TISHRI as f64)
+            + (libm::floor((1.0 / 19.0) * (235.0 * y as f64 - 234.0)));
+
+        Moment::new(
+            FIXED_HEBREW_EPOCH.to_f64_date() - (876.0 / 25920.0)
+                + months_elapsed * (29.0 + (1.0 / 2.0) + (793.0 / 25920.0)),
+        )
+    }
+
+    // ADAR = 12, ADARII = 13
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2217
+    #[allow(dead_code)]
+    fn last_month_of_hebrew_year(h_year: i32) -> u8 {
+        if Self::is_leap_year(h_year) {
+            ADARII
+        } else {
+            ADAR
+        }
+    }
+
+    // Sabbatical Year of the Hebrew Calendar
+    // Lisp code refernence: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2224
+    #[allow(dead_code)]
+    fn hebrew_sabbatical_year(h_year: i32) -> bool {
```

Comment:
suggestion: rename to say `is_...`, since this is a function returning a boolean

Reply:
Makes sense, thank you.